chainntnfs/txconfnotifier.go

```diff
@@ -297,6 +322,42 @@ func (tcn *TxConfNotifier) ConnectTip(blockHash *chainhash.Hash,
 		}
 	}
 
+	// In order to update the height hint for all the required transactions
+	// under one DB transaction, we'll gather the set of unconfirmed
+	// transactions along with the ones that confirmed at the current
+	// height.
+	var txs []chainhash.Hash
+	for confirmedTx := range tcn.txsByInitialHeight[tcn.currentHeight] {
+		txs = append(txs, confirmedTx)
+	}
+out:
+	for maybeUnconfirmedTx := range tcn.confNotifications {
+		for height, confirmedTxs := range tcn.txsByInitialHeight {
+			// Skip the transactions that confirmed at the current
+			// height as those have already been added.
+			if height == tcn.currentHeight {
```

Comment:
I think using `blockHeight` makes it more clear that we are looking at the height of the new connected block.